federatedscope/core/configs/config.py

```diff
@@ -238,37 +239,42 @@ def freeze(self, inform=True):
         """
         self.assert_cfg()
         self.clean_unused_sub_cfgs()
-        # save the final cfg
-        with open(os.path.join(self.outdir, "config.yaml"), 'w') as outfile:
-            from contextlib import redirect_stdout
-            with redirect_stdout(outfile):
-                tmp_cfg = copy.deepcopy(self)
-                tmp_cfg.cfg_check_funcs.clear()
-                print(tmp_cfg.dump())
-            if self.wandb.use:
-                # update the frozen config
-                try:
-                    import wandb
-                    import yaml
-                    cfg_yaml = yaml.safe_load(tmp_cfg.dump())
-                    wandb.config.update(cfg_yaml, allow_val_change=True)
-                except ImportError:
-                    logger.error(
-                        "cfg.wandb.use=True but not install the wandb package")
-                    exit()
-
-        if inform:
+        if save:  # save the final cfg
+            Path(self.outdir).mkdir(parents=True, exist_ok=True)
+            with open(os.path.join(self.outdir, "config.yaml"),
+                      'w') as outfile:
+                from contextlib import redirect_stdout
+                with redirect_stdout(outfile):
```

Comment:
@yxdyc What's the reason to write files via redirecting the standard output stream?

Reply:
No special considerations, just personal convention, and I tested that the way it has been written makes the later `yaml.safe_load()` work correctly.  Perhaps changing the code here into sth. with `yaml.safe_dump()` would make it look more elegant.